### ISSUE TITLE
what is the state of the project 

### ISSUE THREAD

**Issue by awebdeveloper** [pojala/electrino#20]
@pojala what is state of this project. Is it active. Even if you can list out the help you need some of us can help you out.

**Comment by gochev** [comment]
seeing the last commits it doesnt look like that.

**Comment by morajabi** [comment]
This is an amazing project :(

**Comment by postadelmaga** [comment]
it is but it should have some basic functionality in order to be used in some project, such as:
 - windows/osx/linux support
 - notifications 

With these in place I think you can write some basic app

**Comment by norman784** [comment]
maybe we need to shift our attention to other attempts like [https://github.com/jscherer92/Quark](https://github.com/jscherer92/Quark)

**Comment by morajabi** [comment]
@norman784 Yeah, I agree, just we should think how we can help to get there quicker, as it is a very useful project. Not only one person can get us there, so if any can help, let's do it :)

**Comment by norman784** [comment]
to me the project need a mac and a Linux dev, that can do the window and notification stuff for macOS and Linux, then start working on the common API stuff to execute the javascript (if I'm not wrong in mac is JavascriptCore and Linux dunno what library could be used).

Also Linux will be trickier because will be more like electron, because linux itself has no webkit/chakra like library installed by default.

I was thinking that for macOS the project need to be objective-c based because the common c++ API could be used without issues. I'm not Linux expert so dunno how to proceed in that platform.

For now I'm not committing to  help in anything because don't have time right now, but I'm interested in the project itself.

**edit:** we could continue the discussion in their repo.

**Comment by postadelmaga** [comment]
@norman784 
I am not an expert about C libraries but I am a linux user and I am sure about the following points:
 - webkit was developed by KDE folks (KDE/QT is one of the main linux project)  so the package/library is widely available in every Linux distribution 
 - linux have powerful system package manager that manage dependency 

So basically there is no problem in relation to webkit ( i am not sure what chakra is ):  
On linux you just need to set the right dependency once you create a package and the package manager will take care of installing it.

something not very important now because:
- there isn't an actual working built for linux :)
- usually the maintainer of the package is in charge of creating it ( not the developers )

What I can suggest is to use Cross-platform window library [QT](https://www.qt.io), to me, is the best candidate:
At the end of the day, it think, you just need to create a window with a webview inside. 

p.s.
don't get it wrong QT is free if you use for non commercial project ...

**Comment by norman784** [comment]
Check out whats [Chakra](https://github.com/Microsoft/ChakraCore), for linux maybe will be the best option to go with QT, but the whole thing about electrino was a lightweight app (in size, because the performance will be the same I think), and using Chakra on Windows/Webkit on Mac you save a lot, and you don't need to distribute the binaries, right now a simple Hello World app in Electron is about 40mb, and with the electrino philosophy will be just a few Kb or Mb.

But at the end we need to try different alternatives and see whats more useful and easy to implement.

**Comment by awebdeveloper** [comment]
http://www.i-programmer.info/news/167-javascript/5418-javascript-to-be-the-default-langauge-for-gnome.html

**Comment by postadelmaga** [comment]
@norman784  ok got you, I thought Chakra was a OSX library ... 
... so even better I can see Chakra is available for Linux too and Arch linux folks has already their build: https://aur.archlinux.org/packages/chakracore/

**Comment by norman784** [comment]
@awebdeveloper sounds cool I really don't follow too much Linux so that was new to me.

@postadelmaga Oh! didn't know that, also could be nice to use the same engine in at least 2 platforms, so if Chakra already runs on Linux could be a better solution, or maybe in Gnome environments  use their build-in javascript engine.

**Comment by postadelmaga** [comment]
@norman784 
Sticking it to Gnome only would be too limiting in my opinion: the linux usershare is already low so if you filter it by DE gonna be even lower.

I would go for something that not force you to use a DE instead of another, so far Chakra looks best candidate.

**Comment by norman784** [comment]
we are discussing in other [threat/repo](https://github.com/jscherer92/Quark/issues/3) about almost the same and found out that also nodejs has a [chakra implementation](https://github.com/nodejs/node-chakracore) maintained by the same developers.

**Comment by Mayeu** [comment]
You all may also be interested by the [Yue project](https://github.com/yue/yue) :)

**Comment by styfle** [comment]
I made a list of similar frameworks here: https://github.com/styfle/awesome-desktop-js

**Comment by mainrs** [comment]
@styfle make sure to add it to @sindresorhus list of awesome lists :)

**Comment by styfle** [comment]
@SirWindfield I will eventually but I need to create a better contributing.md file.
He is quite strict with those lists as you can see with my other list here: https://github.com/sindresorhus/awesome/pull/1124

**Comment by piyush1104** [comment]
@styfle lol, I came to this repo only through your list.
